### PR TITLE
[14.0][ADD] sale_barcode: add module

### PR DESCRIPTION
**Pull request by dessanhemrayev**
This module adds the product to the sales order line according to the barcode information provided by the barcode/datamatrix device.

### CODE REVIEW

**Comment by dessanhemrayev** [comment]
Create lot and/or select lot if used in combination with sale_order_lot_selection feature is not implemented yet

File: sale_barcode/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        product = self._product_barcode_search(barcode)
+        if not product:
+            raise UserError(_("There is no product with such a barcode"))
+        self.order_line = [(0, 0, {"product_id": product.id})]
+        return {"type": "ir.actions.act_window_close"}
+
+    @api.model
+    def _product_barcode_search(self, barcode):
+        """Returns the found product by barcode"""
+        return self.env["product.product"].search([("barcode", "=", barcode)], limit=1)
```
Comment (aayartsev) [inline sale_barcode/models/sale_order.py]: Do we need to write limit=1 when we have same constraint in model?

**Review by aayartsev — CHANGES_REQUESTED** [review]

File: sale_barcode/models/sale_order.py (hunk 1 above)
Comment (dessanhemrayev) [reply sale_barcode/models/sale_order.py]: Thanks) I'll fix it

**Review by aayartsev — APPROVED** [review]

**Review by aayartsev — APPROVED** [review]
It looks good

**Pull request closed without merging**